docs/paradigm-layouts.md

```diff
@@ -0,0 +1,223 @@
+as a **wordform cell**.
+
+A wordform cell is either a static wordform cell or a dynamic wordform
+cell. A _dynamic_ wordform cell contains exactly one `${lemma}`
+placeholder. The placeholder must be **filled** by an external system,
+and the cell may be substituted with a generated wordform. A _static_
+wordform cell does not contain a placeholder, and can be displayed to
+users verbatim, without any substitution.
+A paradigm layout is a _dynamic layout_ when it contains one or more
+_dynamic wordform cells_. A paradigm layout is a _static layout_ when
+it contains zero _dynamic_ wordform cells. A _dynamic layout_ **MUST**
+be _filled_ before the layout is ready for presentation.
+
+Aside from the presence or absence of the `${lemma}` placeholder, the
+exact syntax of a wordform cell is defined by the implementation.
+
+### Partial Grammar
+
+Here is a partial W3C EBNF grammar of the layout file specification.
+The syntax of `WordformCell` and `Tag` are implementation-dependent.
+
+```ebnf
+Layout ::= (Row NL)+
+
+Row ::= BlankRow | HeaderRow | ContentRow
+
+HeaderRow ::= HeaderLabel (TAB EmptyCell)*
+ContentRow ::= ContentCell (TAB ContentCell)*
+
+ContentCell ::= RowLabel
+              | ColumnLabel
+              | MissingForm
+              | EmptyCell
+              | WordformCell
+BlankRow ::= EmptyCell (TAB EmptyCell)*
+
+HeaderLabel ::= ('#' SP Tag)+
+RowLabel ::= ('_' SP Tag)+
+ColumnLabel ::= ('|' SP Tag)+
+MissingForm ::= '-' '-'
+EmptyCell ::= SP*
+
+TAB ::= #x09
+NL  ::= #x0A
+SP  ::= #x20
+```
+
+### Example
+
+Here is a **static paradigm layout file** with one **pane**. Each
+**row** has row labels, with two **tags** each. Each column has
+a **column label**, with some having one tag, and some having two tags.
+There are a three **missing forms** under the `All` column.
+
+Derived from: <https://en.wikipedia.org/wiki/Swahili_grammar#Personal_pronouns>
+
+```tsv
+| Class	| Ind	| Comb | Suffix	| Gen | Suffix	| All
+_ 1 _ Sg	mimi	-mi	_angu	--
+_ 2 _ Sg	wewe	-we	-ako	--
+_ 3 _ Sg	yeye	-ye	-ake	--
+_ 1 _ Pl	sisi	-si	-etu	sisi
+_ 2 _ Pl	nyinyi	-nyi	-enu	nyote
+_ 3 _ Pl	wao	-o	-ao	wote
+```
+
+
+[vd]: https://www.visidata.org/
+[pycharm-tsv]: https://www.jetbrains.com/help/pycharm/editing-csv-and-tsv-files.html
+
+
+Where to place paradigm layout files
+------------------------------------
+
+Paradigm layout files are placed in the following directory structure
+template:
+
+```
+layouts
+├── {paradigm-name}
+│   ├── {size-1}.tsv
+│   ├── {size-2}.tsv
+│   └── {size-3}.tsv
+├── {paradigm-name}.tsv
+└── {paradigm-name}.tsv
+```
+
+Files directly within the `layouts` directory, can be:
+
+ * a `.tsv` layout file, whose filename stem (part before the `.tsv`) is
+   the _paradigm name_ that this layout corresponds to; or
+ * a subdirectory, whose filename is the _paradigm name_ this directory
+   corresponds to. Within this subdirectory are `.tsv` layout files that
+   indicate the different _size options_ available for this _paradigm
```

Comment:
Might be good to briefly explain the fact that paradigms can come in different sizes towards the top too, maybe under **Components of the paradigm system**.